### DISCUSSION TITLE
Dependant Parameters

### DISCUSSION THREAD

**Discussion by Ncoder2** [pydantic/pydantic#4078] [Question]
Hi Team, 
I am using fastapi, pydantic to perform request validation. I wanted to understand about a particular scenario and see if its possible  with pydantic.

Below is the json request format in the request:
The "type" parameter below can be "electric" or "fuel":

if its electric, the input would be:

`{
"type": "electric",
"input": {
"charge_capacity": "7.2kW"
}
}
`

if its fuel, the input would be:

`{
"type": "fuel",
"input": {
"fuel_capacity": "30L"
}
}
`

In summary, the parameter within "input" will change based on the type parameter, if the requirement is to just have one API to address both the case, can i validate the input parameter to be based on the value of the "type" parameter ? 

I should be able to validate, if the type is "electric", then "charge_capacity" should be present, else if type is "fuel" then "fuel_capacity" should be present

**Comment by samuelcolvin** [comment]
You need a discriminated union, https://pydantic-docs.helpmanual.io/usage/types/#discriminated-unions-aka-tagged-unions

**Reply by Ncoder2** [reply]
Thank you @samuelcolvin ! i think this should do the trick, at the same time i wanted to understand, if i passed this as my request class and if i needed to access just the input as a dict/json, would i be able to do it directly, or should i use something like json.dumps(request.input) ?

**Reply by samuelcolvin** [reply]
Yes, or you can use "exclude"
